From f94d07c07d98812e140462ce638ae8a1f419dc15 Mon Sep 17 00:00:00 2001
From: Zhipeng Gong <zhipeng.gong@intel.com>
Date: Wed, 21 Mar 2018 13:34:09 +0800
Subject: [PATCH 0691/1240] [UPSTREAM] drm/i915/gvt: nop user interrupt

i915 adds MI_USER_INTERRUPT for each request although requests from
shadow context already add MI_USER_INTERRUPT, which causes some
unecessary interrupt handling. NOP MI_USER_INTERRUPT instruction to
save some cpu cycles.
---
 drivers/gpu/drm/i915/gvt/cmd_parser.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/gvt/cmd_parser.c b/drivers/gpu/drm/i915/gvt/cmd_parser.c
index 3fc6672..fd6a8ce 100644
--- a/drivers/gpu/drm/i915/gvt/cmd_parser.c
+++ b/drivers/gpu/drm/i915/gvt/cmd_parser.c
@@ -1089,6 +1089,7 @@ static int cmd_handler_mi_user_interrupt(struct parser_exec_state *s)
 {
 	set_bit(cmd_interrupt_events[s->ring_id].mi_user_interrupt,
 			s->workload->pending_events);
+	patch_value(s, cmd_ptr(s, 0), MI_NOOP);
 	return 0;
 }
 
-- 
2.7.4

